test/run_tests.sh

```diff
@@ -4,4 +4,5 @@ export KB_DEPLOYMENT_CONFIG=$script_dir/../deploy.cfg
 export KB_AUTH_TOKEN=`cat /kb/module/work/token`
 export PERL5LIB=$script_dir/../lib:$PATH:$PERL5LIB
 cd $script_dir/../test
-perl -e 'opendir my $dh, "."; my @l = grep { /\\.pl$/ } readdir $dh; foreach my $s (@l) { print("Running ".$s."\n"); system "perl", $s; }'
+perl -e 'opendir my $dh, "."; my @l = grep { /\.pl$/ } readdir $dh; foreach my $s (@l) { print("Running ".$s."
+"); system "perl", $s; }'
```

Comment:
Perl tests should be called using `prove` (this is an existing error, but might as well fix it), and perl test files should have the suffix `.t`, not `.pl`. I suggest editing this file to have this content:

``` sh
#!/bin/bash
echo "Running $0 with args $@"
if [ -L $0 ] ; then
script_dir=$(cd "$(dirname "$(readlink $0)")"; pwd -P) # for symbolic link
else
script_dir=$(cd "$(dirname "$0")"; pwd -P) # for normal file
fi
base_dir=$(cd $script_dir && cd .. && pwd);
export KB_DEPLOYMENT_CONFIG=$script_dir/../deploy.cfg
export KB_AUTH_TOKEN=`cat /kb/module/work/token`
echo "Removing temp files..."
rm -rf /kb/module/work/tmp/*
echo "Finished removing temp files."
export PERL5LIB=$base_dir/lib:$PERL5LIB
cd $base_dir
prove -lvrm $script_dir
```

and rename the test scripts to have the extension `.t`, instead of `.pl`. The perl test harness can then work its magic properly!

Reply:
This suggestion could best be made to the kb-sdk project.  As I know the run_tests.sh file was automatically generated when a kb-sdk module is created.

Reply:
I've got a PR in the works to put this change into `kb-sdk`. :)  However, that won't affect existing projects like this one, where the Makefile has already been generated (running `make` generates the `run_tests.sh` and some other scripts). Ideally you would edit the Makefile so that it generates the appropriate `run_tests.sh` script; here is a version that creates the correct scripts: https://github.com/kbaseapps/RAST_SDK/blob/use_cpanfile_for_deps/Makefile